### PR TITLE
feat: Keep previous pagination size after remove similarity filter

### PR DESCRIPTION
**Pull request by leiyre**
# Description

This PR allows to disable pagination for similarity search keeping the previous pagination size

**Type of change**

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

**How Has This Been Tested**

Please describe the tests that you ran to verify your changes. And ideally reference `tests`.

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [ ] follows the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I added comments to my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: frontend/models/Dataset.js [hunk 1]
```diff
@@ -79,7 +79,11 @@ class ObservationDataset extends Model {
   }
 
   get visibleRecords() {
-    return this.results.records.slice(0, this.viewSettings.pagination.size);
+    if (this.viewSettings.pagination.disabledPagination) {
+      return this.results.records;
+    } else {
+      return this.results.records.slice(0, this.viewSettings.pagination.size);
```
Comment (frascuchon) [inline frontend/models/Dataset.js]: Is there a condition in the app where the size of the record list is not the same as the pagination size?

If not, could we just show the `results.records` list without the slice?

File: frontend/database/modules/datasets.js [hunk 2]
```diff
@@ -466,26 +466,26 @@ async function _updateTaskDataset({ dataset, data }) {
 }
 
 async function _updatePagination({ id, size, page }) {
-  const updatedSize = getSizeRecords(size);
+  //const updatedSize = getSizeRecords(size);
```
Comment (frascuchon) [inline frontend/database/modules/datasets.js]: Please, remove commented lines. We can always roll them back by using git history.

File: frontend/database/modules/datasets.js [hunk 3]
```diff
@@ -466,26 +466,26 @@ async function _updateTaskDataset({ dataset, data }) {
...
+//   const isWeakLabelingView = $nuxt.$route.query.viewMode === "labelling-rules";
+//   const isSimilaritySearch = $nuxt.$route.query.vectorId;
+//   const sizeIfSimilaritySearchisActivate = 50;
+//   const updatedSize =
+//     !isWeakLabelingView && isSimilaritySearch
+//       ? sizeIfSimilaritySearchisActivate
+//       : size;
+
+//   return updatedSize;
+// };
```
Comment (frascuchon) [inline frontend/database/modules/datasets.js]: Please, remove commented lines. We can always roll them back by using git history.

File: frontend/models/Dataset.js (hunk 1 above)
Comment (leiyre) [reply frontend/models/Dataset.js:85]: oops yes

File: frontend/database/modules/datasets.js (hunk 3 above)
Comment (leiyre) [reply frontend/database/modules/datasets.js]: oops yes

**Review by frascuchon — APPROVED** [review]
Tested in dev and it works like a charm!

File: frontend/models/Dataset.js (hunk 1 above)
Comment (leiyre) [reply frontend/models/Dataset.js]: yes, I think was old logic

**Pull request merged**